### PR TITLE
Pass functions UI

### PR DESCRIPTION
**Pull request by aditiii23**

### CODE REVIEW

File: frontend/src/components/GeneratePass/GeneratePass.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    try {
+      setIsLoading(true)
+      if (Object.keys(err).length < 1) {
+        let res = await axios.post(`${apiUrl}/passes/generatePass/`, newPass, {
+          headers: {
+            authorization: "Bearer " + localStorage.getItem("token"),
+          },
+        })
+        if (res.data.success) {
+          localStorage.setItem("newPass", JSON.stringify(res.data.newPass))
```
Comment (mrpriyansh) [inline frontend/src/components/GeneratePass/GeneratePass.js]: What do you want to do with this line?

File: frontend/src/components/Signup/Signup.js [hunk 2]
```diff
@@ -17,6 +18,8 @@ const Signup = () => {
     password: "",
     confirmpwd: "",
   })
+  const [isLoading, setIsLoading] = useState(false)
```
Comment (mrpriyansh) [inline frontend/src/components/Signup/Signup.js:21]: Why these changes are visible here??

File: frontend/src/components/ViewPasses/ViewPasses.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    fetchPasses()
+  }, [])
+
+  return (
+    <>
+      <div className={viewPassesStyle.container}>
+        {data?.map((item) => (
+          <div className={viewPassesStyle.card}>
+            <Pass
+              key={item?.id}
```
Comment (mrpriyansh) [inline frontend/src/components/ViewPasses/ViewPasses.js:32]: Do you know why key is used?

**Review by mrpriyansh — CHANGES_REQUESTED** [review]

File: frontend/src/components/ViewPasses/ViewPasses.js (hunk 3 above)
Comment (aditiii23) [reply frontend/src/components/ViewPasses/ViewPasses.js:32]: Key is used basically to give an identity to the elements. Like here, each pass id is the key for each pass. This is for unique identification for each pass.

File: frontend/src/components/Signup/Signup.js (hunk 2 above)
Comment (aditiii23) [reply frontend/src/components/Signup/Signup.js:21]: Will figure out and tell you..(after going back from class)

File: frontend/src/components/GeneratePass/GeneratePass.js (hunk 1 above)
Comment (aditiii23) [reply frontend/src/components/GeneratePass/GeneratePass.js]: Store the new pass data in the string form in the local storage.

Comment (mrpriyansh) [reply frontend/src/components/GeneratePass/GeneratePass.js]: Are you using the stored pass anywhere?

**Review by mrpriyansh — CHANGES_REQUESTED** [review]

File: frontend/src/components/GeneratePass/GeneratePass.js (hunk 1 above)
Comment (aditiii23) [reply frontend/src/components/GeneratePass/GeneratePass.js]: No...

**Review by mrpriyansh — APPROVED** [review]

**Pull request merged**